sympy/concrete/summations.py

```diff
@@ -1535,10 +1545,18 @@ def get_residue_factor(numer, denom, alternating):
         residues = [residue(residue_factor, z, root) for root in nonint_roots]
         return -S.Pi * sum(residues)
 
-    if not (a.is_finite and b is S.Infinity):
-        return None
+    rational_function_symmetry = get_function_symmetry(numer, denom)
+    if not (rational_function_symmetry == 'even'):
```

Comment:
Rather than `if not (a == b)` you can write `if a != b`.

Actually though since inside here you alread have an `if rational_function_symmetry == 'odd'` it would be better to just use `if/elif/else`.